code/src/reducers/reusable.js

```diff
@@ -0,0 +1,26 @@
+import { gameReducer } from './gameReducer'
+import { ui } from './ui'
+
+export const HandleMove = (direction) => {
+  return (dispatch, getState) => {
+    dispatch(ui.actions.setLoading(true))
+    const player = getState().gameReducer.playerName
+    const body = JSON.stringify({
+      username: player,
+      type: 'move',
+      direction
+    })
+    fetch('https://wk16-backend.herokuapp.com/action', {
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/json'
+      },
+      body
+    })
+      .then((res) => res.json())
+      .then((json) => {
+        dispatch(gameReducer.actions.makeMove(json))
```

Comment:
Same thing about thunks

Reply:
Move all fetch into the reducer, using thunk. Fixed.